library/ssl_tls13_client.c

```diff
@@ -2647,6 +2647,42 @@ static int ssl_tls13_handshake_wrapup(mbedtls_ssl_context *ssl)
 
 #if defined(MBEDTLS_SSL_SESSION_TICKETS)
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+/* From RFC 8446 section 4.2.10
+ *
+ * struct {
+ *     select (Handshake.msg_type) {
+ *         case new_session_ticket:   uint32 max_early_data_size;
+ *         ...
+ *     };
+ * } EarlyDataIndication;
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_parse_early_data_ext(mbedtls_ssl_context *ssl,
+                                          const unsigned char *buf,
+                                          const unsigned char *end)
+{
+    /* Make sure early data indication extension is received from
+     * NewSessionTicket. */
+    if (!mbedtls_ssl_is_handshake_over(ssl)) {
+        return MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER;
+    }
```

Comment:
```suggestion
```
I do not think we need that. It seems that the intent is to check that this function is not called when parsing a message different from NewSessionTicket that may contain an early data extension. In that case I prefer to make it clear in the function name and its documentation. Regarding the function name `ssl_tls13_parse_nst_early_data_ext()` (as before) or even `ssl_tls13_parse_new_session_ticket_early_data_ext()`. With that I hope that if someone thinks about using it in another context this will ring a bell.